apps/docs/pages/guides/auth/social-login/auth-google.mdx

```diff
@@ -3,119 +3,215 @@ import Layout from '~/layouts/DefaultGuideLayout'
 
-- A box will appear called `OAuth client created`.
-- Copy and save the values under `Your Client ID` and `Your Client Secret`.
+Unlike the OAuth flow which requires the use of a web browser, the native Sign in with Google flow on Android uses the [operating system's built-in functionalities](https://developers.google.com/android/reference/com/google/android/gms/auth/api/identity/package-summary) to prompt the user for consent. Note that native sign-in has been rebranded as _One Tap sign-in on Android_ by Google, which you should not confuse with _One Tap sign in for web_, as mentioned below.
 
-## Enter your Google credentials into your Supabase Project
+When the user provides consent, Google issues an identity token (commonly abbreviated as ID token) that is then sent to your project's Supabase Auth server. When valid, a new user session is started by issuing an access and refresh token from Supabase Auth.
 
-<SocialProviderSettingsSupabase provider="Google" />
+If you're building an Android app using Flutter, you can use `signInWithGoogle()` to quickly get the user's consent and sign them in to your Supabase project:
 
-## Add login code to your client app
+```dart
+final AuthResponse = await supabase.auth.signInWithGoogle();
+```
 
-When your user signs in, call [signInWithOAuth()](/docs/reference/javascript/auth-signinwithoauth) with `google` as the `provider`:
+In the Supabase JavaScript library, which you can use with web-based native frameworks like React Native or Expo, you can invoke this functionality like so:
 
-```js
-async function signInWithGoogle() {
-  const { data, error } = await supabase.auth.signInWithOAuth({
-    provider: 'google',
-  })
-}
+```ts
+await supabase.auth.signInWithIdToken({
+  provider: 'google',
+  token: '<identity token received from the OS>',
+  access_token: '<access token received from the OS>'
+})
 ```
 
-You can view the full list of query parameters and their descriptions [here](https://developers.google.com/identity/protocols/oauth2/web-server#creatingclient).
+Please take a look at these open-source projects which may help you obtain an ID token directly from the OS:
+
+- [react-native-google-signin/google-signin](https://github.com/react-native-google-signin/google-signin).
 
-When your user signs out, call [signOut()](/docs/reference/javascript/auth-signout) to remove them from the browser session and any objects from localStorage:
+For apps written in Kotlin, please consult the [community maintained library](/docs/reference/kotlin/introduction).
 
-```js
-async function signout() {
-  const { error } = await supabase.auth.signOut()
+### Configuration
+
+1. Configure OAuth credentials for your Google Cloud project in the [Credentials](https://console.cloud.google.com/apis/credentials) page of the console. When creating a new OAuth client ID, choose _Android_ for the application type. Use the instructions on screen to provide the SHA-1 certificate fingerprint used to sign your Android app.
```

Comment:
Here as well, we could probably rephrase it to something like "choose which ever platform you want to add Google sign in to, Android or iOS for the application type." to acknowledge the iOS Google login.